src/NUnitFramework/framework/Guard.cs

```diff
@@ -20,8 +21,12 @@ internal static class Guard
         /// <param name="name">The name of the argument</param>
         public static void ArgumentNotNull([NotNull] object? value, string name)
         {
-            if (value == null)
-                throw new ArgumentNullException(name, "Argument " + name + " must not be null");
+            if (value is null)
```

Comment:
generally is better to use `is null`/`is not null` which translates to `ReferenceEquals` it's guaranteed to be fast and does not enter equality operator overloads which are allowed to return whatever they like `(new Car() == null) == true)`.

Reply:
It was on my list TODO to replace all ` == null` with ` is null` and all ` != null` with ` is not null` once I finished completing nullability.